### PR TITLE
[Enhancement] Add predicate when calculating iceberg table statistics

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        public IcebergContext(Types.StructType schema) {
+            this.schema = schema;
+        }
+
+        public Types.StructType getSchema() {
+            return schema;
+        }
+    }
+
+    public Expression convert(List<ScalarOperator> operators, IcebergContext context) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:58]: Any old UT we can use for this new logic?

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:46]: No, I have added ut about this pr.

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by Youngwb — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 74 / 123 (60.16%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[885]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|0|6|00.00%|[253, 254, 255, 256, 257, 260]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|0|5|00.00%|[388, 389, 390, 392, 394]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|3|00.00%|[141, 142, 145]|
|:large_blue_circle:|com/starrocks/connector/iceberg/cost/IcebergTableStatisticCalculator.java|2|5|40.00%|[60, 68, 217]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|2|5|40.00%|[121, 122, 123]|
|:large_blue_circle:|com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java|69|97|71.13%|[77, 78, 79, 107, 108, 110, 112, 113, 115, 122, 129, 142, 146, 162, 170, 188, 199, 204, 210, 218, 220, 224, 226, 228, 233, 244, 250, 255]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/InPredicateOperator.java|1|1|100.00%|[]|

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by Youngwb — DISMISSED** [review]

**Review by caneGuy — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 74 / 124 (59.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|2|00.00%|[891, 892]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|0|6|00.00%|[253, 254, 255, 256, 257, 260]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|0|5|00.00%|[388, 389, 390, 392, 394]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|3|00.00%|[141, 142, 145]|
|:large_blue_circle:|com/starrocks/connector/iceberg/cost/IcebergTableStatisticCalculator.java|2|5|40.00%|[60, 68, 217]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|2|5|40.00%|[121, 122, 123]|
|:large_blue_circle:|com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java|69|97|71.13%|[77, 78, 79, 107, 108, 110, 112, 113, 115, 122, 129, 142, 146, 162, 170, 188, 199, 204, 210, 218, 220, 224, 226, 228, 233, 244, 250, 255]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/InPredicateOperator.java|1|1|100.00%|[]|

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by Youngwb — DISMISSED** [review]

**Review by caneGuy — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 79 / 129 (61.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|2|00.00%|[891, 892]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|0|6|00.00%|[253, 254, 255, 256, 257, 260]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|0|5|00.00%|[388, 389, 390, 392, 394]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|3|00.00%|[141, 142, 145]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|2|5|40.00%|[121, 122, 123]|
|:large_blue_circle:|com/starrocks/connector/iceberg/cost/IcebergTableStatisticCalculator.java|3|6|50.00%|[63, 68, 217]|
|:large_blue_circle:|com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java|73|101|72.28%|[78, 79, 80, 108, 109, 111, 113, 114, 116, 123, 131, 145, 149, 165, 174, 193, 204, 209, 215, 223, 225, 229, 231, 233, 238, 249, 255, 261]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/InPredicateOperator.java|1|1|100.00%|[]|

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 78 / 127 (61.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[947]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|0|6|00.00%|[254, 255, 256, 257, 258, 259]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|0|5|00.00%|[388, 389, 390, 392, 394]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|3|00.00%|[141, 142, 145]|
|:large_blue_circle:|com/starrocks/connector/iceberg/cost/IcebergTableStatisticCalculator.java|2|5|40.00%|[58, 66, 215]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|2|5|40.00%|[121, 122, 123]|
|:large_blue_circle:|com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java|73|101|72.28%|[79, 80, 81, 109, 110, 112, 114, 115, 117, 124, 132, 146, 150, 166, 175, 194, 205, 210, 216, 224, 226, 230, 232, 234, 239, 250, 256, 262]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/InPredicateOperator.java|1|1|100.00%|[]|

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by caneGuy — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        @Override
+        public String visit(ScalarOperator scalarOperator, Void context) {
+            return scalarOperator.accept(this, context);
+        }
+
+        public String visitVariableReference(ColumnRefOperator operator, Void context) {
+            return operator.getName();
+        }
+
+        public String visitConstant(ConstantOperator operator, Void context) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:283]: visitCastOperator

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java]: done

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by Youngwb — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 74 / 126 (58.73%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[947]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|0|6|00.00%|[254, 255, 256, 257, 258, 259]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|0|5|00.00%|[361, 362, 363, 365, 367]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|3|00.00%|[141, 142, 145]|
|:large_blue_circle:|com/starrocks/connector/iceberg/cost/IcebergTableStatisticCalculator.java|2|5|40.00%|[58, 66, 215]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|2|5|40.00%|[121, 122, 123]|
|:large_blue_circle:|com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java|69|100|69.00%|[79, 80, 81, 109, 110, 112, 114, 115, 117, 124, 131, 144, 148, 164, 172, 190, 201, 206, 212, 221, 230, 232, 236, 238, 240, 245, 256, 263, 268, 277, 285]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/InPredicateOperator.java|1|1|100.00%|[]|

**Review by caneGuy — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by stephen-shelby** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**